internal/manifestwork/manifestwork.go

```diff
@@ -0,0 +1,120 @@
+package manifestwork
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	reflect "reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	workv1 "open-cluster-management.io/api/work/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+)
+
+//go:generate mockgen -source=manifestwork.go -package=manifestwork -destination=mock_manifestwork.go
+
+type ManifestWorkCreator interface {
+	GarbageCollect(ctx context.Context, clusters clusterv1.ManagedClusterList, mcm kmmv1beta1.ManagedClusterModule) error
+	SetManifestWorkAsDesired(ctx context.Context, mw *workv1.ManifestWork, mcm kmmv1beta1.ManagedClusterModule) error
+}
+
+type manifestWorkGenerator struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func NewCreator(client client.Client, scheme *runtime.Scheme) ManifestWorkCreator {
+	return &manifestWorkGenerator{
+		client: client,
+		scheme: scheme,
+	}
+}
+
+func (mwg *manifestWorkGenerator) GarbageCollect(ctx context.Context, clusters clusterv1.ManagedClusterList, mcm kmmv1beta1.ManagedClusterModule) error {
+	manifestWorks, err := mwg.getOwnedManifestWorks(ctx, mcm)
+	if err != nil {
+		return fmt.Errorf("failed to get owned ManifestWorks: %w", err)
+	}
+
+	clusterNames := map[string]string{}
```

Comment:
I think using the [`sets.String`](https://pkg.go.dev/k8s.io/apimachinery@v0.25.4/pkg/util/sets#String) type from apimachinery would be more explicit here.

Reply:
Awesome! Couldn't make more sense :+1: